openstates/pr/committees.py

```diff
@@ -1,77 +1,111 @@
 # -*- coding: utf-8 -*-
-import lxml.html
-import lxml.etree
+import re
+
 import os
-from billy.scrape import NoDataForPeriod
+
 from billy.scrape.committees import CommitteeScraper, Committee
 from billy.scrape.utils import convert_pdf
+from openstates.utils import LXMLMixin
 
-import re
 
 def clean_spaces(s):
     """ remove \xa0, collapse spaces, strip ends """
     if s is not None:
-        return re.sub('\s+', ' ', s.replace(u'\xa0', ' ')).strip()
+        return " ".join(s.split())
 
 
-class PRCommitteeScraper(CommitteeScraper):
+class PRCommitteeScraper(CommitteeScraper, LXMLMixin):
     jurisdiction = 'pr'
     latest_only = True
 
-    def scrape(self, term, chambers):
-        if 'upper' in chambers:
+    def scrape(self, chamber, term):
+        if chamber == 'upper':
             self.scrape_upper()
-        elif 'lower' in chambers:
+        elif chamber == 'lower':
             self.scrape_lower()
 
     def scrape_upper(self):
-        raise Exception('needs to be rewritten for 2013')
+        url = 'http://senado.pr.gov/comisiones/Pages/default.aspx'
+        doc = self.lxmlize(url)
+        for link in doc.xpath('//a[contains(@href, "ComposicionComisiones")]/@href'):
+            doc = self.lxmlize(link)
+            (pdf_link, ) = doc.xpath('//a[contains(@href,".pdf")]/@href')
+            self.scrape_upper_committee(pdf_link)
 
     def scrape_upper_committee(self, url):
         filename, resp = self.urlretrieve(url)
-        root = lxml.etree.fromstring( convert_pdf(filename,'xml'))
-        for link in root.xpath('/pdf2xml/page'):
-            comm = None
-            for line in link.findall('text'):
-                text = line.findtext('b')
-                if text is not None and text.startswith('Comisi'):
-                    comm = Committee('upper',text);
+        lines = convert_pdf(filename, 'text').split('\n')
+        comm = None
+        comm_name = ''
+        title = ''
+        for line in (x.decode('utf8') for x in lines):
```

Comment:
so, while I agree with this change (of course); we should likely file an issue on `convert_pdf` to accept an `encoding=` kwarg, so this isn't needed (unicode sandwich)